crypto/crypto-derive/src/hasher.rs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+/// Converts a camel-case string to snake-case
+pub fn camel_to_snake(text: &str) -> String {
+    let mut out = String::with_capacity(text.len());
+    let mut first = true;
+    text.chars().for_each(|c| {
+        if !first && c.is_uppercase() {
+            out.push('_');
```

Comment:
Note that a name like `foo_Bar` will be `foo__bar` (two _ in a row). Is this the expected behavior?

Reply:
Since the name of the static variable for a certain hasher is pretty inconsequential, and implicit to the user I'm wary of making this function a bit too clever for its own good.